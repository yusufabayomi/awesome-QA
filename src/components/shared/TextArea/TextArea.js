import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string.isRequired,
  input: PropTypes.object,
};

const TextArea = ({ label, input }) => {
  return (
    <>
      <label className='form-label'>{label}</label>
      <textarea {...input}></textarea>
    </>
  );
};

TextArea.propTypes = propTypes;
export default TextArea;
