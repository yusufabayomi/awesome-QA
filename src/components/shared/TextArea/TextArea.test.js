import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TextArea from './TextArea';

afterEach(cleanup);

describe('Renders TextArea Component', () => {
  const props = {
    label: 'Answer',
  };

  it('renders correctly', () => {
    const { asFragment } = render(<TextArea {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
