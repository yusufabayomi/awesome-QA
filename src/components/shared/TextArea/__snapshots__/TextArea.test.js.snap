// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TextArea Component renders correctly 1`] = `
<DocumentFragment>
  <label
    class="form-label"
  >
    Answer
  </label>
  <textarea />
</DocumentFragment>
`;
